From 1a30b123bab5f8f4683b3ecd735ac536067aea2b Mon Sep 17 00:00:00 2001
From: Xiaotian Wu <wuxiaotian@loongson.cn>
Date: Wed, 17 May 2023 17:22:49 +0800
Subject: [PATCH] Add some definitions for LoongArch

Added CHAR8, CHAR16, WCHAR and VOID type definitions for LoongArch
---
 inc/loongarch64/efibind.h | 12 +++++++-----
 1 file changed, 7 insertions(+), 5 deletions(-)

diff --git a/inc/loongarch64/efibind.h b/inc/loongarch64/efibind.h
index aaf3fb7..8ed83a5 100644
--- a/inc/loongarch64/efibind.h
+++ b/inc/loongarch64/efibind.h
@@ -42,9 +42,10 @@ typedef int64_t             intptr_t;
 // Basic EFI types of various widths
 //
 
-#ifndef __WCHAR_TYPE__
-# define __WCHAR_TYPE__ short
-#endif
+#include <stddef.h>
+
+typedef wchar_t CHAR16;
+#define WCHAR CHAR16
 
 typedef uint64_t   UINT64;
 typedef int64_t    INT64;
@@ -54,12 +55,13 @@ typedef int32_t    INT32;
 
 typedef uint16_t   UINT16;
 typedef int16_t    INT16;
+
 typedef uint8_t    UINT8;
+typedef char       CHAR8;
 typedef int8_t     INT8;
-typedef __WCHAR_TYPE__ WCHAR;
 
 #undef VOID
-#define VOID    void
+typedef void       VOID;
 
 typedef int64_t    INTN;
 typedef uint64_t   UINTN;
